### PR TITLE
Modify middleware order and add backchannel injection sample

### LINKED ISSUE #1430: Adding custom data to channel data.
Since the webchat has migrated to V4 I have not found any documentation to add custom data into messages using the embedded webchat.

My sample originally used the following code.

    const speechOptions = {
        speechRecognizer: new BotChat.Speech.BrowserSpeechRecognizer(),
        speechSynthesizer: new BotChat.Speech.BrowserSpeechSynthesizer()
    }

    var dl = new BotChat.DirectLine({ secret: 'YOUR_SECRET_HERE' });

    BotChat.App({
        botConnection: Object.assign({}, dl, {
            postActivity: activity => {
                var newActivity = Object.assign({}, activity, { ChannelData: { 'Channel': 'WebChat'} });
                return dl.postActivity(newActivity);
            }
        }),
        user: { id: 'userid' },
        bot: { id: 'botid' },
        resize: 'window',
        speechOptions: speechOptions
    }, document.getElementById("bot"));

Is there anyway to do something similar along the above line for V4?

### PR DESCRIPTION
**Pull request by compulim**
Resolves #1430 
# Background

We want to enable custom middleware to modify existing behavior of saga. For example,

- Add email to every `DIRECT_LINE/POST_ACTIVITY`
- Automatically downscale photos

In order to enable this work, we need to tweak the middleware order. Moving custom middleware up will enable more scenarios, and should not disable any existing scenarios (because it is already at the bottom of the stack).

The sample would modify Redux action using custom middleware, and fed into saga for further processing (i.e. send it out).

Will fix #1430.

# CHANGELOG.md

### Changed
- Core: Saga will run after custom middleware, in [#1331](https://github.com/Microsoft/BotFramework-WebChat/pull/1331)
   - Custom middleware run before saga to allow user to modify default behavior

### Fixed
- Build: Webpack watch mode now emit non-minified code for shorter dev RTT, in [#1331](https://github.com/Microsoft/BotFramework-WebChat/pull/1331)

### Samples
- Backchannel: Inject custom data into every `POST_ACTIVITY`, in [#1331](https://github.com/Microsoft/BotFramework-WebChat/pull/1331)

### CODE REVIEW

**Comment by compulim** [comment]
Adding @corinagum for code review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +41,14 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
       - `webpack@4.24.0`
       - `webpack-command@0.4.2`
 
+### Fixed
+- Build: Webpack watch mode now emit non-minified code for shorter dev RTT, in [#1331](https://github.com/Microsoft/BotFramework-WebChat/pull/1331)
```
Comment (corinagum) [inline CHANGELOG.md]: *emits

File: samples/backchannel-inject-data-on-post/index.html [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<!DOCTYPE html>
+<html lang="en-US">
+  <head>
+    <title>Web Chat: Inject data on post activity</title>
+    <!--
+      For demonstration purpose, we are using development branch of Web Chat at "/master/webchat.js".
```
Comment (corinagum) [inline samples/backchannel-inject-data-on-post/index.html]: Please copy from latest full-bundle sample -- fixed verbiage

Comment (Raziel0714) [reply samples/backchannel-inject-data-on-post/index.html]: Hi , I tried but didn't find the sample for 'Inject data on post activity', Could I have a link here? Thank you!

Comment (Raziel0714) [reply samples/backchannel-inject-data-on-post/index.html]: Hi, if i send this custom data to bot, how can i retrieve the message from bot side? Thank you.

**Pull request merged**